compaction/compaction.cc

```diff
@@ -549,6 +549,15 @@ class compaction {
                         _table_s.get_compaction_strategy().adjust_partition_estimate(_ms_metadata, _estimated_partitions, _schema));
     }
 
+    double average_estimate_of_droppable_tombstone_ratio() const {
+        double total_estimate = 0;
+        auto compaction_time = gc_clock::now();
+        for (const auto& sst : _sstables) {
+            total_estimate += sst->estimate_droppable_tombstone_ratio(compaction_time, _table_s.get_tombstone_gc_state(), _schema);
+        }
+        return total_estimate / _sstables.size();
+    }
```

Comment:
let's do calculation in compaction::setup(). also good for not recomputing it on every sstable creation (list can grow to more than 1k elements). sstable list _sstables is not immutable, might have sstables removed from it during compaction so we might lose info. Also, we shouldn't account for fully expired sstables, that might be skipped completely by compaction, as they won't be accounted for in partition estimation. Look for '_estimated_partitions += sst->get_estimated_key_count();'.

Reply:
Done.